draft-ietf-mls-protocol.md

```diff
@@ -1723,10 +1723,10 @@ struct {
 
         case commit:
           Commit commit;
-          opaque confirmation_tag<0..255>;
     }
 
     opaque signature<0..2^16-1>;
+    optional<MAC> confirmation_tag;
```

Comment:
An alternative would be to use:
select (MLSPlaintext.content_type) {
        case commit:
          MAC confirmation_tag;
}
sparing the need for the definition below that the confirmation tag must be present iff it is a commit.

Reply:
Not a bad idea; it would save an octet and give you "syntactic" enforcement.  I'm still inclined to go with the `optional<>` approach for two reasons:

1. I'm not aware of TLS syntax implementatiosn that implement `select()` in any sort of automated fashion, so there would be manual logic here in any case.  It's just slightly easier to omit with `optional<>`.

2. If we do `select` for `confirmation_tag`, we should probably also do it with `membership_tag` (selecting based on `sender_type`), which gets to be ugly.